spec/integration/riak/search_spec.rb

```diff
@@ -0,0 +1,104 @@
+require 'spec_helper'
+require 'riak/search'
+
+describe 'Object-oriented Search API', test_client: true, integration: true, search_config: true do
+  before :all do
+    create_index
+    configure_bucket
+    load_corpus
+  end
+
+  let(:term){ 'bitcask' }
+
+  describe 'queries' do
+    let(:index){ Riak::Search::Index.new test_client, index_name }
+    subject { Riak::Search::Query.new test_client, index, term }
+
+    it 'performs queries' do
+      results = nil
+      expect{ results = subject.results }.to_not raise_error
+      expect(results.raw).to_not be_empty 
+      expect(results).to_not be_empty
+    end
+
+    it 'performs limited and sorted queries' do
+      subject.rows = 5
+      results = nil
+      expect{ results = subject.results }.to_not raise_error
+      expect(results.raw).to_not be_empty
+      expect(results.length).to eq 5
+    end
+  end
+
+  describe 'results from queries' do
+    let(:index){ Riak::Search::Index.new test_client, index_name }
+    let(:query){ Riak::Search::Query.new test_client, index, term }
+    subject { query.results }
+    
+    it 'exposes search-result documents' do
+      expect(subject).to_not be_empty
+
+      expect(subject.docs).to_not be_empty
+      expect(first = subject.docs.first).to be
+
+      expect(first.score).to be_a Numeric
+
+      expect(first.bucket_type).to be_a Riak::BucketType
+      expect(first.bucket).to be_a Riak::Bucket
+      expect(first.key).to be_a String
+    end
+
+    it 'exposes RObjects' do
```

Comment:
Right, I'm not arguing whether it's useful to have such an accessor (I agree on that point). I'm arguing against the user paying the cost of fetching that object or datatype up-front, by default. Consider that they might want to display a list of search results directly out of the index, using stored fields; forcing their search result to fetch each object in the list adds substantial latency. Also, since the `ResultCollection#each` method calls `[]` (which returns the RObject), rather than yielding each document, they can't opt-out as easily.